src/core/public/entry_point.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * This is the entry point used to boot the frontend when serving a application
+ * that lives in the Kibana Platform.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { CoreSystem } from './core_system';
+
+const injectedMetadata = JSON.parse(
+  document.querySelector('kbn-injected-metadata')!.getAttribute('data')!
+);
+
+if (process.env.IS_KIBANA_DISTRIBUTABLE !== 'true' && process.env.ELASTIC_APM_ACTIVE === 'true') {
+  // @ts-ignore
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const { init } = require('@elastic/apm-rum');
```

Comment:
1 optional: can be loaded lazily
2 if the package throws an exception, the platform doesn't handle it. shouldn't it be inlined with i18n loading mechanism?

Reply:
> can be loaded lazily

It's wrapped in a 'static' `if env` block, meaning that webpack will drop the whole code block (and therefor remove the dependant apm module from the tree) during compilation when the condition is false.

Reply:
I forgot this is the client code. But it's even worse, as we have different artifacts for different environments 🙈 I'd prefer we have a truly immutable configurable bundle. and adjust the use of `apm-run`  via the config.

Reply:
> But it's even worse, as we have different build artifacts for different environments

IDK, it's quite a common practice to use that kind of branch cutting trick to reduce bundle size when a specific import is unused in some env, however it's supposed to be mostly used to avoid increasing the prod bundle size, and here, the branch is actually removed in dev instead of prod, so maybe you're right and this is an unnecessary risk/optimization.

Reply:
>  however it's supposed to be mostly used to avoid increasing the prod bundle size, and here, the branch is actually removed in dev instead of prod

I don't think that's right, it is excluded from prod: `IS_KIBANA_DISTRIBUTABLE !== 'true'`.

I don't have a strong opinion here, but I was leaning towards keeping this synchronous rather than introducing a top-level await (I haven't tested to see if our tooling even supports that yet).

Reply:
> I don't think that's right, it is excluded from prod

I probably have to go back to conditionals 101.